src/extension/background-script/index.ts

```diff
@@ -76,7 +76,15 @@ const handleInstalled = (details) => {
 
 // listen to calls from the content script and calls the actions through the router
 // returns a promise to be handled in the content script
-const routeCalls = (message, sender) => {
+const routeCalls = (
```

Comment:
`type` was replaced by `action`. Should be possible to remove `type`.  

To keep things aligned maybe try:

```ts
import type { MessageDefault } from "~/types";

interface Message extends MessageDefault {
  args?: unkown;
}

...

const routeCalls = (
  message: Message
  ...
```

Usually `args` is an object I think